src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/ProvideApplicationPartFactoryAttribute.cs

```diff
@@ -11,15 +12,37 @@ namespace Microsoft.AspNetCore.Mvc.ApplicationParts
     [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false)]
     public sealed class ProvideApplicationPartFactoryAttribute : Attribute
     {
+        /// <summary>
+        /// Creates a new instance of <see cref="ProvideApplicationPartFactoryAttribute"/> with the specified type.
+        /// </summary>
+        /// <param name="factoryType">The factory type.</param>
         public ProvideApplicationPartFactoryAttribute(Type factoryType)
         {
             ApplicationPartFactoryType = factoryType ?? throw new ArgumentNullException(nameof(factoryType));
         }
 
+        /// <summary>
+        /// Creates a new instance of <see cref="ProvideApplicationPartFactoryAttribute"/> with the specified type name.
+        /// </summary>
+        /// <param name="factoryTypeName">The assembly qualified type name.</param>
         public ProvideApplicationPartFactoryAttribute(string factoryTypeName)
         {
+            if (string.IsNullOrEmpty(factoryTypeName))
+            {
+                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(factoryTypeName));
+            }
+
+            ApplicationPartFactoryTypeName = factoryTypeName;
         }
 
+        /// <summary>
+        /// The factory type.
+        /// </summary>
         public Type ApplicationPartFactoryType { get; }
+
+        /// <summary>
+        /// THe factory type name.
+        /// </summary>
+        public string ApplicationPartFactoryTypeName { get; }
```

Comment:
I would suggest not exposing either of these properties, and having a `GetFactoryType` method instead. It's a little misleading to expose the properties because only one will be set for a given use case.